### ISSUE TITLE
ColorHelper 3.0 Showing Previews for SCSS/SASS variables that are the same as an HTML color name

### ISSUE THREAD

**Issue by mrwweb** [facelessuser/ColorHelper#170]
## Description

After upgraded to ColorHelper 3.0, I see the following in a `.scss` file:

![An SCSS variable $blue includes has the HTML color blue's swatch after the $](https://user-images.githubusercontent.com/1238696/111000854-a8997000-8337-11eb-9c0b-a446f784d20a.png)

`$blue` is an SCSS/SASS variable for a color and is not a reference to the HTML color "blue":

![Definition of $blue variable shows correct color](https://user-images.githubusercontent.com/1238696/111000902-bbac4000-8337-11eb-8201-bdfcd40a0119.png)

This was correctly _not_ highlighted in previous versions. It would be awesome if the correct color was shown (#142), but I'd be more than happy to just not have this highlighted.

## Support Info

- ST ver.: 3211
- Platform: windows
- Arch: x64
- Plugin ver.: 3.0.0
- Install via PC: True
- coloraide ver: 0.1.0a7
- mdpopups ver.: 4.0.2
- markdown ver.: 3.2.2
- pygments ver.: 2.1a0
- jinja2 ver.: 2.10.1

## Steps to Reproduce Issue

Use any SCSS variable with a name that matches a named HTML color as a CSS property value.

![](https://user-images.githubusercontent.com/1238696/111001235-45f4a400-8338-11eb-9bb6-5b6eec62196f.png)

```SCSS
color: $blue;
color: $tomato;
color: $rebeccapurple;
color: $darkgoldenrod;
```

**Comment by facelessuser** [comment]
@mrwweb Please provide what syntax package you use for your SCSS. As you can see below, it can vary depending on what SCSS file you use:

<img width="148" alt="Screen Shot 2021-03-12 at 3 15 54 PM" src="https://user-images.githubusercontent.com/1055125/111004293-fd43e780-8345-11eb-9ebc-a2a51c04f4b5.png">

I didn't go through and test every package out there that does SCSS, and the ones I did test, I probably only made a passing glance to make sure it was responding to colors.

**Comment by facelessuser** [comment]
Oh, never mind, I see we were talking about the variables on the right hand side.

**Comment by facelessuser** [comment]
I have a fix 🙁.

<img width="204" alt="Screen Shot 2021-03-12 at 3 21 29 PM" src="https://user-images.githubusercontent.com/1055125/111004669-ac80be80-8346-11eb-974b-3dace41bc971.png">

We just need to make sure our trigger for color words doesn't have a `$` before it:

```
(?:\b(?<![-#&$])[\w]{3,}(?!\()\b
```

**Comment by mrwweb** [comment]
Glad you figured it out! That change makes sense to me and I can't think of any obvious reason why that would have unintended consequences. In fact, I assume the old version must have worked that way?

**Comment by facelessuser** [comment]
> Glad you figured it out! That change makes sense to me and I can't think of any obvious reason why that would have unintended consequences. In fact, I assume the old version must have worked that w

Yeah, it is one of those things that I missed in the rewrite. We want to exclude it from looking like SCSS variables, CSS variables, HTML entities, functions calls, etc. Part of that gets filtered out by scope rules, but some syntax files don't have a resolution that makes that easy, so we also try to do a few lookbehinds and lookaheads on color words. Color words are honestly the trickiest. Things like `rgba(...)` are far less likely to give you a false positive.

This made me double-check, and I put in a fix for color words with a trailing `-` as they may be part of a CSS variable and such.

**Comment by mrwweb** [comment]
Awesome. That all makes sense! Glad this was an easy fix :) Congrats on getting 3.0 out the door! It seems pretty great so far.

**Comment by facelessuser** [comment]
Thanks! If all the missed bugs are just like this, I'll be pretty happy.
